### PR TITLE
[chore] add chloggen for changelog

### LINKED ISSUE #5647: Update the changelog procedure to match contrib
The contrib repo has updated its changelog procedure to reduce friction for PRs / users. This issue is an open question whether or not the core repo should follow the same process.

This would ensure users contributing to both repos have a consistent experience and makes the release process consistent as well.

### PR DESCRIPTION
**Pull request by codeboten**
This change updates the changelog process in this repo to match the process in the contrib repo.

Fixes #5647

### CODE REVIEW

**Review by jpkrohling — COMMENTED** [review]
Looks good, but perhaps there's something small missing?

```
$ make chlog-update VERSION=v0.60.0
cd ./internal/tools && go install go.opentelemetry.io/build-tools/chloggen
chloggen update -version v0.60.0
FAIL: update: expected one instance of <!-- next version -->

make: *** [Makefile:458: chlog-update] Error 1
```

**Review by codeboten — COMMENTED** [review]
@jpkrohling I was waiting for the release to be done before updating the changelog & updating the files in the `unreleased` directory

**Review by djaglowski — APPROVED** [review]
LGTM, but consider adding [this section](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-a-changelog-entry) to `CONTRIBUTING.md`.

**Comment by djaglowski** [comment]
> I found very bad naming 'unreleased' in contrib. It sounds to me like an unreleased code. Can we have different name or under another directory in docs?

I can see why it sounds like actual code would be contained there. The word came directly out of the previous version of the changelog (ie. Unreleased changes). I considered `changelog/` but thought `unreleased` was better because it does not imply that it would contain the complete history of changes. Maybe `.unreleased` or `unreleased_log`?

The placement at the root of the repo was with developer experience in mind. I thought this would make it easy to find and therefore easy to adapt to the new process. Less important on core where we have fewer contributors though.

**Comment by TylerHelmuth** [comment]
I'd vote for `unreleased_log` (or `unreleased_logs` or `unrelease_chlog`) in both repos.  Something that indicates these are future changelog entries.

**Comment by codeboten** [comment]
👍🏻 im ok with a hidden .chloggen directory. It should be easy enough to update chloggen to make this directory configurable

**Comment by codeboten** [comment]
@djaglowski @bogdandrutu PTAL at https://github.com/open-telemetry/opentelemetry-go-build-tools/pull/145 to update the default directory to `.chloggen`.

I plan on updating the contrib repo to match `.chloggen` once that change is merged.

**Review by djaglowski — APPROVED** [review]
LGTM

File: .chloggen/TEMPLATE.yaml
Comment (TylerHelmuth) [inline .chloggen/TEMPLATE.yaml:10]: Should we update this comment (and the one in contrib) to note "issue or pull request"?

**Review by TylerHelmuth — APPROVED** [review]

**Comment by codeboten** [comment]
@bogdandrutu @jpkrohling PTAL

File: internal/tools/go.mod
Comment (jpkrohling) [inline internal/tools/go.mod:15]: Should the versions for tools in `go.opentelemetry.io/build-tools` be aligned?

Comment (codeboten) [reply internal/tools/go.mod:15]: Fixed for consistency

**Comment by codeboten** [comment]
Keeping this PR updated is somewhat painful (as it conflicts every time the changelog is updated). I'd like to get approvals from others @open-telemetry/collector-approvers before updating it again. Otherwise we can always close it

File: internal/tools/go.mod
Comment (Aneurysm9) [inline internal/tools/go.mod:14]: These can be updated to `v0.1.0`.

**Review by Aneurysm9 — APPROVED** [review]

**Review by tigrannajaryan — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Pull request merged**
